internal/lsp/config/watcher_test.go

```diff
@@ -0,0 +1,71 @@
+package config
+
+import (
+	"context"
+	"os"
+	"testing"
+	"time"
+)
+
+func TestWatcher(t *testing.T) {
+	t.Parallel()
+
+	tempDir := t.TempDir()
+
+	configFilePath := tempDir + "/config.yaml"
+
+	configFileContents := `---
+foo: bar
+`
+
+	err := os.WriteFile(configFilePath, []byte(configFileContents), 0o600)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	watcher := NewWatcher(&WatcherOpts{ErrorWriter: os.Stderr})
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	go func() {
+		err = watcher.Start(ctx)
+		if err != nil {
+			t.Errorf("failed to start watcher: %v", err)
+		}
+	}()
+
+	watcher.Watch(configFilePath)
+
+	select {
+	case <-watcher.Reload:
+	case <-time.After(100 * time.Second):
```

Comment:
Can we cap this at like 5-10 seconds or something? Anything above that would seem like something we'd want to look into anyway.

Reply:
Oops, that's a mistake. Fixed in d5bee69

Reply:
Made it 100ms like the others in bf48f11

Reply:
Thanks!